include/aos/common/tools/string.hpp

```diff
@@ -306,10 +348,17 @@ class String : public Array<char> {
         Clear();
 
         for (const auto val : src) {
-            const auto digits = ByteToHex(val);
+            char digits[2];
```

Comment:
string.ByteToHex uses resize that has *end() = 0 that results in stack corruption.
It's better use StaticString for the particular case: no need in convertion to use ByteToHex & Instead can be replaced with append.

Reply:
StaticString is not available here.

Reply:
ah, right